### PR TITLE
New framebuffer driver interface and subsystem modifications

### PR DESCRIPTION
**Pull request by AntonKozlov**

### CODE REVIEW

File: src/drivers/video/fb_videomodes.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+#ifndef VIDEO_FB_VIDEOMODES_H_
+#define VIDEO_FB_VIDEOMODES_H_
+
+struct video_resbpp;
+
+struct fb_videomode {
+	const char *name;	/* optional */
+	uint32_t refresh;	/* optional */
```
Comment (abusalimov) [inline src/drivers/video/fb_videomodes.h]: Missing `#include <stdint.h>`

File: src/drivers/video/fb_videomodes.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	uint32_t sync;
+	uint32_t vmode;
+	uint32_t flag;
+};
+
+extern void fb_videomode_to_var(struct fb_var_screeninfo *var,
+		const struct fb_videomode *mode);
+
+extern void fb_var_to_videomode(struct fb_videomode *mode,
+		const struct fb_var_screeninfo *var);
```
Comment (abusalimov) [inline src/drivers/video/fb_videomodes.h]: Missing forward declaration of `struct fb_var_screeninfo`.
As a result this header can't be included from anywhere, I suppose except from where it is included now.

File: src/drivers/video/fb.c [hunk 3]
```diff
@@ -22,125 +24,157 @@
...
+	struct fb_info *ret;
 
-	return registered_fb[num];
+	ret = NULL;
+	mutex_lock(&fb_static);
+	{
+		struct fb_dev *fb_dev;
+		dlist_foreach_entry(fb_dev, &fb_list, link) {
+			if (fb_dev->info.id == id) {
+				ret = &fb_dev->info;
```
Comment (abusalimov) [inline src/drivers/video/fb.c]: Hmm... Shouldn't there be a break after an element is found?

**Comment by AntonKozlov** [comment]
Please, don't delete the branch. There are referencecies from outside

**Pull request merged**